Shared/Samples/Show portal user info/ShowPortalUserInfoView.swift

```diff
@@ -0,0 +1,304 @@
+            // If loading, show that the user profile will display when complete.
+            if model.isLoading {
+                ContentUnavailableView(
+                    "Portal User Information",
+                    systemImage: "person.crop.circle.dashed",
+                    description: Text("Your portal user information will be displayed here.")
+                )
+                // Otherwise show the user information that was loaded.
+            } else {
+                InfoScreen(model: $model)
+            }
+        }
+        .errorAlert(presentingError: $error)
+    }
+    
+    /// Asynchronously loads the portal user information.
+    private func loadUser() async {
+        do {
+            try await model.loadPortalUser()
+        } catch {
+            // If an error occurs, mark loading as true (to reset)
+            // and store the error to present an alert.
+            model.isLoading = true
+            self.error = error
+        }
+    }
+    
+    /// A composed view for handling portal login, logout, and URL input.
+    @ViewBuilder private var portalDetails: some View {
+        PortalDetailsView(
+            url: $model.portalURLString,
+            model: $model,
+            // Update the model when the portal URL is changed by the user.
+            onSetUrl: {
+                model.portalURLString = $0
+            },
+            // Sign out the user and reset state.
+            onSignOut: {
+                Task {
+                    await model.signOut()
+                }
+            },
+            // Load the user's portal data when sign in is triggered.
+            onLoadPortal: {
+                Task {
+                    await loadUser()
+                }
+            }
+        )
+        .frame(maxWidth: 400)
+        // Set up the authenticator when the view appears.
+        .onAppear(perform: model.setAuthenticator)
+        // Clean up authenticator and credentials when the view disappears.
+        .onDisappear {
+            Task {
+                await model.clearAuthenticator()
+            }
+        }
+        // Attach the authenticator to the view for handling authentication challenges.
+        .authenticator(model.authenticator)
+    }
+}
+
+private extension ShowPortalUserInfoView {
+    @MainActor
+    @Observable
+    class Model {
+        /// The authenticator to handle authentication challenges.
+        @ObservationIgnored var authenticator: Authenticator
+        /// The API key to use temporarily while using OAuth.
+        @ObservationIgnored private var apiKey: APIKey?
+        /// The URL string of the portal to connect to.
+        /// Defaults to the main ArcGIS Online portal.
+        var portalURLString: String = "https://www.arcgis.com"
+        /// Stores the current user's data such as username, email, etc.
+        var userData: PortalUser?
+        /// Stores the information related to user's portal.
+        var portalInfo: PortalInfo?
```

Comment:
Consider renaming `userData`:

```suggestion
        /// Stores the information related to user's portal.
        var portalInfo: PortalInfo?
        /// Stores the current user's data such as username, email, etc.
        var portalUser: PortalUser?
```